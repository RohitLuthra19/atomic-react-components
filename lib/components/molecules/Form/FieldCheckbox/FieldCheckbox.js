// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import InputChoice from '../../../atoms/InputChoice';
import FieldError from '../../../atoms/FieldError';
import styles from './FieldCheckbox.style';

type Props = {
  id?: string,
  className?: string,
  name: string,
  fieldProps?: Object,
};

const FieldCheckbox = ({ id, className, name, fieldProps, ...others }: Props): Node => (
  <div className={className}>
    <Field
      name={name}
      render={({ field }) => <InputChoice type="checkbox" id={id || name} {...field} {...others} />}
      {...fieldProps}
    />
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

FieldCheckbox.defaultProps = {
  id: '',
  className: '',
  fieldProps: {},
};

export default styled(FieldCheckbox)`
  ${styles};
`;
export { FieldCheckbox as FieldCheckboxVanilla };
