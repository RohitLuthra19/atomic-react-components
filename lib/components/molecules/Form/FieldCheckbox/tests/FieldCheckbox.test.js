import React from 'react';

import { shallowWithTheme } from '../../../../../../__mock__/themeMock';
import FieldCheckbox from '../FieldCheckbox';

describe('FieldCheckbox Component', () => {
  let checkBoxWrapper = '';
  beforeEach(() => {
    checkBoxWrapper = shallowWithTheme(<FieldCheckbox />);
  });

  test('should render correctly', () => {
    expect(checkBoxWrapper).toMatchSnapshot();
  });
});
