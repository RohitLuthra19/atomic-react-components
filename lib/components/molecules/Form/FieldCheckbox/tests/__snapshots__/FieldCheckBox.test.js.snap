// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldCheckbox Component should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FieldCheckbox-sc-1ia5ini-0",
        "isStatic": false,
        "rules": Array [
          "margin-bottom:1rem;",
          ";",
        ],
      },
      "displayName": "FieldCheckbox",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FieldCheckbox-sc-1ia5ini-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
